modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItReplicaLifecycleTest.java

```diff
@@ -716,6 +716,55 @@ void testStableAreWrittenAfterRestartAndConcurrentStableUpdate(TestInfo testInfo
         assertTrue(waitForCondition(() -> getNode(0).replicaManager.isReplicaStarted(partId), 10_000L));
     }
 
+    @Test
+    public void testTableEstimatedSize(TestInfo testInfo) throws Exception {
+        startNodes(testInfo, 3);
+
+        Assignment replicaAssignment = (Assignment) calculateAssignmentForPartition(
+                nodes.values().stream().map(n -> n.name).collect(toList()), 0, 1, 1).toArray()[0];
+
+        Node node = getNode(replicaAssignment.consistentId());
+
+        placementDriver.setPrimary(node.clusterService.topologyService().localMember());
+
+        createZone(node, "test_zone", 1, 1);
+        int zoneId = DistributionZonesTestUtil.getZoneId(node.catalogManager, "test_zone", node.hybridClock.nowLong());
+
+        long key = 1;
+
+        {
+            createTable(node, "test_zone", "test_table");
```

Comment:
I'd say that we need a zone with several partitions and couple of tables with different amount of data, e.g.
```
z1 -> t1 -> p1 -> 1 record
          t1->  p2 -> 2 records
          t2 -> p1 -> 4 records
          t2 -> p2 -> 5 records
```
And thus we may assert that t1 contains 3 records, and t2 9 recods.
I do understand that such assertions will mainly check original estimatedSize logic and node the proposed one in given request, however it worth almost nothing to adjust the test in that way and it'll bring us a little more confidence.

Reply:
:ok_hand: